MdpItem.pm

```diff
@@ -1493,6 +1493,80 @@ sub GetItemCover {
     return $seq;
 }
 
+sub ProcessOwnerIds {
+    my $self = shift;
+
+    my $id = $self->GetId();
+    my $root = $self->_GetMetsRoot();
+    my $source_mets = $root->findvalue(q{//METS:fileGrp[@USE='source METS']//METS:FLocat/@xlink:href});
+    my $zipfile = $self->Get('zipfile');
+
+    my $seq_map = {};
+    my $ownerid_map = {};
+
+    if (defined $zipfile && defined $source_mets) {
+        my $file_sys_location = $self->Get( 'filesystemlocation' );
+
+        my $source_METS_filename = Utils::Extract::extract_file_to_temp_cache($id, $file_sys_location, $source_mets);
+
+        # If there is a Google METS, read and parse and get OWNERID
+        # attribute value
+
+        if ($source_METS_filename) {
+            my $metsXmlRef = Utils::read_file($source_METS_filename, 1);
+            if ($$metsXmlRef) {
+                my $parser = XML::LibXML->new();
+                my $tree = $parser->parse_string($$metsXmlRef);
+                my $root = $tree->getDocumentElement();
+
+                foreach my $file ( $root->findnodes(qq{//METS:file[\@OWNERID]}) ) {
+                    my $ownerid = $file->getAttribute('OWNERID');
+                    next unless ( $ownerid );
+                    my @ownerids = split(/ /, $ownerid);
+                    my $fileid = $file->getAttribute('ID');
+                    my $seq = $root->findvalue(qq{//METS:structMap[\@TYPE="physical"]//METS:div[METS:fptr[\@FILEID="$fileid"]]/\@ORDER});
+                    $$seq_map{$seq} = $ownerids[0];
```

Comment:
@aelkiss would the most recent ownerid be listed first or last?

Reply:
There will only be one `OWNERID` on `//METS:file`, which represents the `OWNERID` of the page scan actually in use.

In looking at an example (chi.082891291) I see:

```xml
<METS:file ID="IMG00000001" MIMETYPE="image/jp2" SEQ="00000001" CREATED="2014-10-06T00:00:00" SIZE="409054" CHECKSUM="93a5b01fb43835d0d1a283660108cbe0" CHECKSUMTYPE="MD5" OWNERID="13510798903268757-5" ADMID="IMAGE_METHOD0 ADD_PAGE_DATA_IMG00000001">
        <METS:FLocat xlink:href="00000001.jp2" LOCTYPE="OTHER" OTHERLOCTYPE="SYSTEM"/>
      </METS:file>
```
Other candidate `OWNERID`s that are believed to be the same page will be listed in the referenced `techMD`:

```xml
    <METS:techMD ID="ADD_PAGE_DATA_IMG00000001">
      <METS:mdWrap LABEL="candidates" MDTYPE="OTHER" OTHERMDTYPE="Google">
        <METS:xmlData>
          <gbs:pageNo>PP1</gbs:pageNo>
          <gbs:scanIDwrap>
            <gbs:ownerID>13510798903268757-5</gbs:ownerID>
          </gbs:scanIDwrap>
        </METS:xmlData>
      </METS:mdWrap>
    </METS:techMD>
```

I need to dig up an example, but I believe there will just be a sequence of `gbs:ownerID` elements that lists all candidates.

So, I think the way to get other OWNERIDs that should map to this sequence as well is to look for an element referenced by the `ADMID` that contains an `mdWrap[@LABEL="candidates"]` (and sadly I don't think there's a way to tokenize those IDREF values directly via XPath 1.0, which is all libxml2 supports) and then get all its `gbs:ownerID` descendants -- or equivalently to find the page a specific OWNERID should reference, find the `gbs:candidate` element that contains it, then find the file that references the enclosing techMD...

Reply:
@aelkiss changed the code to collect all the `gbs:ownerID` values that belong to a `techMD` element before processing the `file` elements.